src/Chainweb/RestAPI/Utils.hs

```diff
@@ -137,6 +142,27 @@ type family ReassocBranch (a :: s) (b :: [Type]) :: Type where
     ReassocBranch a '[] = a
     ReassocBranch a (b ': rest) = a :> ReassocBranch b rest
 
+newtype DuplicateHeader = DuplicateHeader HTTP.HeaderName
+instance Show DuplicateHeader where
+    show (DuplicateHeader hn) = "Attempted to set a duplicate header with this name: " <> show hn
+instance Exception DuplicateHeader
+
+addHeader :: HTTP.Header -> [HTTP.Header] -> [HTTP.Header]
+addHeader (hn, hv) hs =
+    (hn, hv) : [ if n == hn then throw (DuplicateHeader hn) else h | h@(n, _) <- hs ]
```

Comment:
I think `find` is more readable/clear than this list comprehension
```suggestion
    case find ((== hn) . fst) hs of
      Just _ -> throw (DuplicateHeader hn)
      Nothing -> (hn, hv) : hs
```

Reply:
(Alternatively with `isJust`)

Reply:
That would be clearer, I did this for extra laziness but it doesn't seem worth it now.